modules/end-to-end/pages/building-tekton-tasks.adoc

```diff
@@ -0,0 +1,298 @@
+
+
+[NOTE] 
+====
+* Each task and task version in the application should be represented by a component, each component should contain the pipeline in use (*tekton-bundle-builder-oci-ta*) and the context to the task within the GH repository (*tasks/echo/0.1* for example).
+* The Component's naming convention is <task_name>-v<task_version> (echo-v01)
+====
+
+Once these resources are added, a PR with *on-pr* and *push* pipelineruns will be sent to your repository.
+
+[NOTE] 
+====
+* The *on-pr* pipelinerun will be created with the context we  have defined for the component and with *on-cel-expression* to run the pipelinerun only in case a change was detected in the component, for example:
+
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request"
+    && target_branch == "main"
+    && (
+    "tasks/echo/0.1/***".pathChanged() || ".tekton/echo-v01-pull-request.yaml".pathChanged()
+    )
+----
+However, when *branch protection* is enabled in your GH repository it might prevent the pipelinerun to run when needed for the branch protection, in this case updating the *on-cel-expression* in the on-pr pipelinerun will solve the issue.
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
+----
+
+* The *Push* pipelinerun is created with a default on-cel-expression.
+For now we need to manually update it to run only if changes were made in the context, example:
+
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "push"
+    && target_branch == "main"
+    && (
+    "tasks/echo/0.1/***".pathChanged() || ".tekton/echo-v01-push.yaml".pathChanged()
+      )
+----
+====
+
+=== Integration tests
+Follow xref:testing:integration/index.adoc[Integration tests] to add integration tests to the application and components.
+
+Example of an *IntegrationTestScenario*:
+[source, yaml]
+----
+apiVersion: appstudio.redhat.com/v1beta2
+kind: IntegrationTestScenario
+metadata:
+  name: my-tasks-enterprise-contract
+  namespace: my-team-tenant
+spec:
+  application: my-tasks
+  contexts:
+    - description: Component testing for echo task version 0.1
+      name: component_echo-v01
+  params:
+    - name: POLICY_CONFIGURATION
+      value: rhtap-releng-tenant/tekton-bundle-standard
+  resolverRef:
+    params:
+      - name: url
+        value: https://github.com/konflux-ci/build-definitions
+      - name: revision
+        value: main
+      - name: pathInRepo
+        value: pipelines/enterprise-contract.yaml
+    resolver: git
+----
+
+[NOTE]
+====
+* Defining *component_<component_name>* will ensure the integration test runs only for this component
```

Comment:
Are you recommending this context? Why should users choose it?

Reply:
See comment above